doc/callbacks.md

```diff
@@ -0,0 +1,484 @@
+Callbacks in mlpack are methods that are called at different states during the
```

Comment:
I suggest changing "different" to "various".  

In this context the word "different" is grammatically imprecise at best, and too overloaded in meaning at worst (eg. "different states" can raise questions like "how are the states different?", or "what are the states different to?", etc).  
Generally the word "different" should only be used to mean "not the same".